pkg/ytconfig/base_generator.go

```diff
@@ -0,0 +1,48 @@
+package ytconfig
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+
+	ytv1 "github.com/ytsaurus/yt-k8s-operator/api/v1"
+)
+
+type BaseGenerator struct {
```

Comment:
Since this `BaseGenerator` is being used as common part for `NodeGenerator` and `Generator` (the second one is for all other components currently) it would be confusing to name it `NodeBaseGenerator`.